### DISCUSSION TITLE
Unable to display images in Flutterflow

### DISCUSSION THREAD

**Discussion by DayneLinx** [AntlerVC/firetable#1156] [Support / Q&A] [answered]
# Rowy Bug: Unable to display images in Flutterflow

## **Describe the bug**

I am having issues displaying images from my Firestore Database that have be input via Rowy. I have been trying to figure this one out and I even created a new Flutterflow instance removing Rowy from the equation. I have no problem displaying images when they have been uploaded using the built in Firestore Data Manager. What I am seeing is that when using a pageview to display the images the indicator is correct in indicating the amount of images that are present in the array. 

## **To Reproduce**

Steps to reproduce the behaviour:

1. Login to my Rowy instance
2. Create a new record
3. Fill in all the relevant details with a selection of data arrays
4. Upload a single image to my ‘featuredImage’ field
5. Upload 3 or more images to my gallery ‘headerImage’ field
6. Log in to Flutterflow
7. Match my DB Schema in collections (Image included)
8. Create the headerImage data type  (image Included) for the images array (Image included)
9. Create a backend query on column view representing my home page (Image included)
10. Defined a parameter on my details page (Image included)
11. Generated children from variables on the pageview widget (Image included)
12. Assigned the ‘downloadURL’ from the Data Structure Field to the pageview (Image included)
13. Ran the app and all that is displayed is the defualt placeholder images from Flutterflow (Image included)
14. Note the Rowy record for Cala Tarida and the details page. The indicators are correct but without images (Image included)
15. Reproduced the above steps without using Rowy and all images in an array displayed without hitch
16. 

## **Expected behaviour**

I expected the images from my firestore database to be displayed instead of the default images.

## **Screenshots**

Please find attached

## **Rowy Run version**

Rowy Run version (Latest as of 03/01/2023 )

## **Desktop (please complete the following information):**

- OS: macOS 12.6.1
- Browser Opera ((Build 21G217) 12.6.1 arm64) / Safari 16.2 (17614.3.7.1.7, 17614)

## **Smartphone (please complete the following information):**

- N/A (Used in a simulator)

## **Additional context**

I am also having troubles retrieving data from string arrays. 
I also see that images regularly failing to load, when I obtain the URL of the image the images are still available in the firestore. (Image included)

Any guidance will be appreciated.

![Screenshot 2023-01-03 at 01 59 49](https://user-images.githubusercontent.com/23277612/210292749-b490ebf8-c630-473a-9e0b-df121b33c1ab.png)
![Screenshot 2023-01-03 at 02 25 20](https://user-images.githubusercontent.com/23277612/210293005-539b38bf-e85b-4224-bdca-70cd1646d4c4.png)

![Screenshot 2023-01-03 at 02 01 00](https://user-images.githubusercontent.com/23277612/210292753-2d343049-8461-4ba0-8bcc-6b6747b6733e.png)
![Screenshot 2023-01-03 at 02 02 13](https://user-images.githubusercontent.com/23277612/210292756-7a967f23-3528-48ea-bf2c-a4682c41f324.png)
![Screenshot 2023-01-03 at 02 07 02](https://user-images.githubusercontent.com/23277612/210292761-d2de3e63-e326-4930-b4cb-76fa00a5ebf4.png)
![Screenshot 2023-01-03 at 02 11 54](https://user-images.githubusercontent.com/23277612/210292762-ed341a8d-7964-491c-b4be-6596f6a32333.png)
![Screenshot 2023-01-03 at 02 12 13](https://user-images.githubusercontent.com/23277612/210292763-c8d27946-e72c-4921-a11e-9e0ebfbf3b7f.png)
![Screenshot 2023-01-02 at 23 45 42](https://user-images.githubusercontent.com/23277612/210292764-a1fc8e0b-6b10-4d74-b344-579fbbefc544.png)

**Comment by BeeBombshell** [comment, marked as answer]
This query has been addressed in the discussion https://github.com/rowyio/rowy/discussions/1085. The discussion describes how users can **extract the download URL** or **an array of download URLs** that can be used to access these images on FlutterFlow
